patch.go

```diff
@@ -176,18 +190,23 @@ func createInitContainerPatch(config agentConfig, createArray bool) patchOperati
 		// *if needed*?
 		Command: []string{"cp", "-v", "-r", config.ArtifactPath, mountPath},
 	}
+
+	if errs := validation.IsDNS1123Label(agentInitContainer.Name); len(errs) != 0 {
+		return patchOperation{}, fmt.Errorf("init container name (%s) is not a valid DNS_LABEL: %v", agentInitContainer.Name, errs)
+	}
```

Comment:
I think it's probably good to have this validation, but the error message might be a little bit cryptic. I think we should also validate the config `parseConfig`, ensuring an image and artifact path are specified. Those are directly under the control of the user.

Reply:
> I think it's probably good to have this validation, but the error message might be a little bit cryptic.

Can you clarify this ? The error message is adapted from the pod specification: https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container . The errs slice should contain a message similar to what it's sent on the kubernetes side (see https://github.com/elastic/apm-mutating-webhook/issues/38):

`a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name', or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')`

>  I think we should also validate the config parseConfig, ensuring an image and artifact path are specified. Those are directly under the control of the user.

Agree, fixed!

Reply:
Sorry, my comment was a little bit cryptic :smile:

What I meant was: the error message talks about the container name being invalid, but the user doesn't directly control the container name -- it's derived from the image name. So by making the error message about the config specified by the user, it should be clearer what they need to change to fix the error.

Reply:
Ah I see, that's a great point! I've updated the error message